external/lzma/Kconfig

```diff
@@ -0,0 +1,11 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config LIBLZMA
+	bool "LZMA library"
+	default y if COMPRESSION_TYPE=1 && BUILD_PROTECTED=n
```

Comment:
1. The default in case `type != 1 || PROTECTED` should be `'n'`, which it will be even if not set in config description.
2. `protected=n` is necessary here because only in flat build case, lzma should compile as part of libexternal.a due to `type == 1`.
In other case (`protected=y`), if `type == 1`, lzma files should be copied to os/compression/ and compile as part of libkernel.a